core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java

```diff
@@ -149,6 +152,59 @@ boolean testClassLoad(final String className, final String asTypeName)
    */
   void waitForBalance() throws AccumuloException;
 
+  /**
+   * Throws an exception if a tablet server can not be contacted.
+   *
+   * @param args
+   *          Command line arguments passed in from user command.
+   * @param secretOption
+   *          Specified instance secret to use for commands.
+   *
+   * @since 2.1.0
+   */
+  boolean fateFail(List<String> args, String secretOption) throws AccumuloException;
+
+  /**
+   * Throws an exception if a tablet server can not be contacted.
+   *
+   * @param args
+   *          Command line arguments passed in from user command.
+   * @param secretOption
+   *          Specified instance secret to use for commands.
+   *
+   * @since 2.1.0
+   */
+  boolean fateDelete(List<String> args, String secretOption) throws AccumuloException;
+
+  /**
+   * Throws an exception if a tablet server can not be contacted.
+   *
+   * @param args
+   *          Command line arguments passed in from user command.
+   * @param filterTxid
+   *          Parsed transaction IDs for print filter.
+   * @param filterStatus
+   *          Parsed TStatus for print filter.
+   * @param secretOption
+   *          Specified instance secret to use for commands.
+   *
+   * @since 2.1.0
+   */
+  String fatePrint(List<String> args, Set<Long> filterTxid, EnumSet<TStatus> filterStatus,
+      String secretOption) throws AccumuloException;
+
+  /**
+   * Throws an exception if a tablet server can not be contacted.
+   *
+   * @param args
+   *          Command line arguments passed in from user command.
+   * @param secretOption
+   *          Specified instance secret to use for commands.
+   *
+   * @since 2.1.0
+   */
+  String fateDump(List<String> args, String secretOption) throws AccumuloException;
```

Comment:
Not sure if we need both "Dump" and "Print" methods. They seem to be doing similar things.

Reply:
Dump provides a lot more information, I will post a snippet below

<details>

<pre>
root@uno> fate dump
[
  {
    "txid": "58136bf11a1bc7d5",
    "stack": [
      {
        "org.apache.accumulo.manager.tableOps.TraceRepo": {
          "traceId": 0,
          "parentId": 0,
          "repo": {
            "org.apache.accumulo.manager.tableOps.create.CreateTable": {
              "tableInfo": {
                "tableName": "test",
                "namespaceId": {
                  "canonical": "+default"
                },
                "timeType": "MILLIS",
                "user": "root",
                "initialTableState": "ONLINE",
                "initialSplitSize": 19,
                "splitFile": "hdfs://groot:8020/accumulo/tmp/fate-58136bf11a1bc7d5/splits",
                "splitDirsFile": "hdfs://groot:8020/accumulo/tmp/fate-58136bf11a1bc7d5/splitsDirs",
                "props": {
                  "table.majc.compaction.strategy.opts.large.compress.threshold": "100M",
                  "table.iterator.majc.vers": "20,org.apache.accumulo.core.iterators.user.VersioningIterator",
                  "table.constraint.1": "org.apache.accumulo.core.data.constraints.DefaultKeySizeConstraint",
                  "table.iterator.scan.vers.opt.maxVersions": "1",
                  "table.iterator.minc.vers": "20,org.apache.accumulo.core.iterators.user.VersioningIterator",
                  "table.majc.compaction.strategy.opts.large.compress.type": "gz",
                  "table.iterator.majc.vers.opt.maxVersions": "1",
                  "table.majc.compaction.strategy": "org.apache.accumulo.tserver.compaction.strategies.BasicCompactionStrategy",
                  "table.iterator.minc.vers.opt.maxVersions": "1",
                  "table.iterator.scan.vers": "20,org.apache.accumulo.core.iterators.user.VersioningIterator",
                  "table.majc.compaction.strategy.opts.filter.size": "250M"
                }
              }
            }
          }
        }
      }
    ]
  
</pre>

</details>

Reply:
Whether or not we want all that information I guess is up for debate.

Reply:
If we had one method that returned the set of transaction IDs, their states, and their blobs like this, code that calls this method can decide what pieces they care about.